streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1140,6 +1140,11 @@ private void completeShutdown(final boolean cleanRun) {
 
         topologyMetadata.unregisterThread(threadMetadata.threadName());
 
+        try {
+            topologyMetadata.unregisterThread(threadMetadata.threadName());
```

Comment:
Oh good catch, I guess that got missed since I ported this fix from [#11601](https://github.com/apache/kafka/pull/11601/) (since that doesn't need to make 0.24) -- fixed